main.go

```diff
@@ -81,11 +79,17 @@ func run(c *cli.Context) error {
 	if err != nil {
 		return errors.Wrapf(err, "failed to parse %s", RefreshInterval)
 	}
-
-	config, err := config.NewConfig(ctx, SubKey, intval, ChannelServerVersion, URLs...)
-	if err != nil {
-		return err
+	if len(SubKeys.Value()) != len(PathPrefix.Value()) {
+		return errors.Errorf("keys-prefix lengths are not equal %s %s %s ", PathPrefix.Value(), SubKeys.Value(), ListenAddress)
 	}
+	configs := map[string]*config.Config{}
+	for index, subkey := range SubKeys.Value() {
+		config, err := config.NewConfig(ctx, subkey, intval, ChannelServerVersion, URLs.Value())
+		configs[PathPrefix.Value()[index]] = config
```

Comment:
We may want to check the error first rather than potentially assign a nil value to the map unnecessarily.